### PR TITLE
chore(common): replace jest with vitest

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What?

This PR replaces jest with vitest.

## Why?

JIRA: SHAPE-7088

### CODE REVIEW

**Comment by eunjae-lee** [comment]
type check files. working on it 🤔

File: packages/demo/package.json [hunk 1]
```diff
@@ -7,7 +7,6 @@
     "dev": "vite",
     "build": "vite build",
     "lint": "eslint .",
-    "test": "jest",
```
Comment (demetriusfeijoo) [inline packages/demo/package.json:10]: hhhh, both the `demo` and `sandbox` packages don't have tests, so no need to keep it here, right? 🤔

**Review by demetriusfeijoo — APPROVED** [review]
Great job and improvement, @eunjae-lee 💯

File: packages/demo/package.json (hunk 1 above)
Comment (eunjae-lee) [reply packages/demo/package.json:10]: yes, that's why i removed them :)

**Pull request merged**